org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/folding/LSPFoldingReconcilingStrategy.java

```diff
@@ -320,4 +306,24 @@ public void setProgressMonitor(IProgressMonitor monitor) {
 	public void initialReconcile() {
 		reconcile(null);
 	}
+
+	@Override
+	public void install(ITextViewer viewer) {
+		if (this.viewer != null) {
+			this.viewer.removeProjectionListener(this);
+		}
+		this.viewer = (ProjectionViewer) viewer;
+		this.viewer.addProjectionListener(this);
+		this.projectionAnnotationModel = this.viewer.getProjectionAnnotationModel();
+	}
+
+	@Override
+	public void uninstall() {
+		setDocument(null);
+		if (viewer != null) {
+			viewer.removeProjectionListener(this);
+			viewer = null;
+		}
+		projectionDisabled();
+	}
```

Comment:
could we move this block back to 178-194 so that the diff looks nicer?

Reply:
fixed